pkg/modifycontroller/modify_volume.go

```diff
@@ -43,35 +43,40 @@ func (ctrl *modifyController) modify(pvc *v1.PersistentVolumeClaim, pv *v1.Persi
 		// First time adding VAC to a PVC
 		return ctrl.validateVACAndModifyVolumeWithTarget(pvc, pv)
 	} else if pvcSpecVacName != nil && curVacName != nil && *pvcSpecVacName != *curVacName {
-		targetVacName := *pvcSpecVacName
-		if pvc.Status.ModifyVolumeStatus != nil {
-			targetVacName = pvc.Status.ModifyVolumeStatus.TargetVolumeAttributesClassName
+
+		// Check if the PVC is in uncertain State
+		pvcKey, err := cache.MetaNamespaceKeyFunc(pvc)
+		if err != nil {
+			return pvc, pv, err, false
 		}
-		if *curVacName == targetVacName {
-			// if somehow both curVacName and targetVacName is same, what does this mean??
-			// I am not sure about this.
```

Comment:
Why was this branch initially included? Will bring up in next SIG Implementation meeting, because this week's iteration was skipped.